### ISSUE TITLE
Security Warning in Mac Mojave

### ISSUE THREAD

**Issue by eric-hemasystems** [yakyak/yakyak#998]
Since upgrading to Mojave I get this warning:

<img width="433" alt="yak-yak-warning" src="https://user-images.githubusercontent.com/34583914/47796222-46a5b600-dcfa-11e8-9372-8b094a4d33c5.png">

Wanted to confirm it is intended and necessary. If so perhaps it can be documented somewhere so folks know it is safe?

Only thing slightly related that I could find is another Electron app that got the same issue. Looks like they were getting it because the app installing itself to launch on startup:

https://github.com/keybase/client/issues/13911

**Comment by emerout** [comment]
Same behavior on Mojave, each time yakyak is starting.

**Comment by averissimo** [comment]
We use this module to create an option allowing  yakyak to start automatically.

We'd welcome a fix *(if it's necessary)* for this as a pull request :)

**Comment by eric-hemasystems** [comment]
Maybe the thing to do is open a PR against `auto-launch`? Close this issue. When `auto-launch` comes up with a more friendly solution yakyak just bumps the dep?

**Comment by averissimo** [comment]
This option is different from all others, as you noticed it runs a check everytime yakyak runs to see if the system is configured to run yakyak at startup *(visual check is based on this)*

The extension we use seems abandoned :\

Can you see if this https://github.com/yakyak/yakyak/releases/tag/v1.5.4-alpha-issue-998 solves this problem?

(basically it uses an alternative method)

**Comment by eric-hemasystems** [comment]
I have confirmed that the alpha release that you linked to solves the problem. I completely removed the version I was using. Confirmed that if I installed it that I got the prompt again. Remove that and installed the one you linked to (built the package locally). I no longer got the warning. Also confirmed that on reboot it is still launching on login.

NOTE: `gulp-livereload` depends on `event-stream` which depended on `flatmap-stream`. That package included rouge code (something related to stealing bitcoins). It was yanked from npm which means I couldn't build your package. Bumping to the latest version of `gulp-livereload` got me going again and removed the exploited code. See https://github.com/dominictarr/event-stream/issues/116 for more info. Since it is a dev package I am guessing it was not in the built packages?

**Comment by averissimo** [comment]
@eric-hemasystems 

Can you just perform one more test? making sure if checking the option to run as startup actually works and then setting it as off

Regarding `gulp-livereload`, we already bumped the package in the master repository.

As it was not included in the binaries *(it is just a development package)* I feel that it doesn't need an emergency release. *(I've confirmed it by looking through the binaries for Linux, windows, osx and deb/rpm.. I don't think I missed it, but I can make mistakes)*

If you feel otherwise I'll do it let me know why and we'll do it.

Anyway, thanks for reporting!! The community is great as 2 users did it *(including you)*

**Comment by eric-hemasystems** [comment]
Just tested and I confirmed that the setting actually works. If I uncheck it it does not start on login, if I check it it does. I can also see it adding and removing the item from my list of apps to start in the Mac settings.

I did have one little issue. At first when I tried to remove it did not remove it and it continued to login after boot. I manually removed it once and after that it started working. I am guessing that my old "start at login" was from the old version of the app so I basically had to manually remove that old setup. But once I had a clean slate it worked perfectly.
